### PR TITLE
Add "Posix" global version conditional for Emscripten targets

### PR DESCRIPTION
**Pull request by ryuukk**
I just had this thought after seeing this: https://forum.dlang.org/post/odqtqqravqlfcqjgfywo@forum.dlang.org

Haven't tested yet, but i submit the PR otherwise i'll forget

And i can't compile ldc from source:

`cmake -G Ninja ../  -DCMAKE_BUILD_TYPE=Release  -DCMAKE_INSTALL_PREFIX=$PWD/../install-ldc`
```
/usr/bin/ld: cannot find -lLLVMSymbolize: No such file or directory
```

### CODE REVIEW

File: driver/main.cpp [hunk 1]
```diff
@@ -752,8 +752,26 @@ void registerPredefinedTargetVersions() {
     VersionCondition::addPredefinedGlobalIdent("D_HardFloat");
     break;
   case llvm::Triple::wasm32:
+    VersionCondition::addPredefinedGlobalIdent("WebAssembly");
+    if (triple.getOS() == llvm::Triple::Emscripten) {
+      // Emscripten reimplements Posix APIs
+      VersionCondition::addPredefinedGlobalIdent("linux");
+      VersionCondition::addPredefinedGlobalIdent("Posix");
+      VersionCondition::addPredefinedGlobalIdent("CRuntime_Musl");
```
Comment (JohanEngelen) [inline driver/main.cpp:760]: is this correct? Musl?  that is more than just posix.

File: driver/main.cpp [hunk 2]
```diff
@@ -752,8 +752,26 @@ void registerPredefinedTargetVersions() {
...
+      VersionCondition::addPredefinedGlobalIdent("X86_Any");
```
Comment (JohanEngelen) [inline driver/main.cpp:761]: X86 for sure is wrong

File: driver/main.cpp [hunk 3]
```diff
@@ -752,8 +752,26 @@ void registerPredefinedTargetVersions() {
     VersionCondition::addPredefinedGlobalIdent("D_HardFloat");
     break;
   case llvm::Triple::wasm32:
+    VersionCondition::addPredefinedGlobalIdent("WebAssembly");
+    if (triple.getOS() == llvm::Triple::Emscripten) {
+      // Emscripten reimplements Posix APIs
+      VersionCondition::addPredefinedGlobalIdent("linux");
```
Comment (JohanEngelen) [inline driver/main.cpp:758]: `linux` probably also incorrect?

File: driver/main.cpp (hunk 1 above)
Comment (kinke) [reply driver/main.cpp:760]: Hmm interesting. Do you have a link to confirm this? I found https://github.com/emscripten-core/emscripten/pull/13006 which seems to support this.

The proper place to add the musl-compatible predefined versions is https://github.com/ldc-developers/ldc/blob/dd0ff1ee838437972ee05a00b691a5e5354f6943/driver/main.cpp#L921-L923.

**Comment by kinke** [comment]
Superseded by #4750.

**Pull request closed without merging**